### PR TITLE
Dev server config

### LINKED ISSUE #4027: Make dev server deployable
Today, we have a server provisioned for use as a dev server. However, deployment for the dev server is not set up. Having a dev server is useful for a few reasons:

- We can show client demos for features which aren't ready to be on staging or prod yet
- Non techies can more easily see proposed changes of a new feature
- Some networking things can be tested more easily on a remote server
- A dev invoking a long running or computationally intensive operation may prefer to do so on a remote environment rather than their local machine

Since we currently only have 1 dev server provisioned, use of the dev server will need to be coordinated in Slack. This is annoying, but imo the Parabol team is small enough to where we can still do this reasonably. In the future, wishlist is to have optional ["review app"](https://about.gitlab.com/blog/2016/11/22/introducing-review-apps/) deployed for any branch which is then torn down when the branch is merged or otherwise destroyed.

#### Acceptance Criteria (optional)
- [ ] any dev on Parabol team can deploy to our dev server
- [ ] dev server can be visited on the internet, for demos and such
- [ ] doc created in Notion for the above

- **Estimated effort:** 8 hours

### PR DESCRIPTION
**Pull request by tiffanyhan**
Resolve #4027

#### Acceptance Criteria (optional)
- [ ] Please follow the steps [here](https://www.notion.so/parabol/How-to-Deploy-to-our-development-server-16b7b2ec90134124ab5f6e2d650ffcf9) to deploy to the development server. Wait for the CI pipeline on the development server to complete successfully.
- [ ] Go to https://action-dev-nyc1-01.parabol.co/ and see changes live, manually test, etc.

### CODE REVIEW

**Comment by nickoferrall** [comment]
> Resolve #4027
> 
> #### Acceptance Criteria (optional)
> * [ ]  Force push to development sever. Watch CI pipeline complete successfully.
> * [ ]  Go to https://action-dev-nyc1-01.parabol.co/ and see changes live, manually test, etc.

Hey @tiffanyhan, I just want to check the steps I need to take to test this. I tried doing the following:
- fetched your branch, created a new branch from it and made a small change to some text
- pushed the code. It successfully passed the CI checks in [Actions](https://github.com/ParabolInc/parabol/actions)
- checked https://action-dev-nyc1-01.parabol.co/me but the small change that I made hasn't updated

I'm running `yarn && yarn dev -i`. Is there anything else I should be doing to push to the dev server and trigger the update?

**Comment by tiffanyhan** [comment]
Ah that must be my fault. I suspect it's something to do with versioning that the app isn't updating. Thank you for testing this @nickoferrall !

**Comment by nickoferrall** [comment]
OK cool - no worries!

**Comment by tiffanyhan** [comment]
Hey @nickoferrall my original AC wasn't clear. Can you please try again? I've updated the AC and it now links to a doc in notion.

**Comment by nickoferrall** [comment]
Thanks for writing that doc @tiffanyhan, that's very helpful of you. The changes are working so I'll go ahead and merge the PR 👍

File: .circleci/config.yml [hunk 1]
```diff
@@ -70,8 +82,6 @@ jobs:
               sudo pip install rethinkdb==2.3.0
               rethinkdb-restore --force packages/cypress/fixtures/rdb_test.tar.gz
               node scripts/toolbox/softenDurability.js
-              yarn db:migrate
```
Comment (mattkrick) [inline .circleci/config.yml:73]: i haven't tested this, but note that these are called after 
`cp $DEVOPS_WORKDIR/environments/cypress .env`
if we execute these commands before copying over the cypress env then we are executing them on a completely different set of environment variables, which probably isn't what you want.

@tiffanyhan worth a hotfix before going to prod?

Comment (tiffanyhan) [reply .circleci/config.yml:73]: shoot thank you for looking into this @mattkrick. i really misinterpreted the sequence of steps here. i'll link you to the hotfix.

**Pull request merged**